ambry-account/src/test/java/com/github/ambry/account/MockHelixAccountServiceFactory.java

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.commons.MockHelixPropertyStore;
+import com.github.ambry.commons.Notifier;
+import com.github.ambry.config.HelixPropertyStoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.manager.zk.ZkBaseDataAccessor;
+import org.apache.helix.manager.zk.ZkClient;
+import org.apache.helix.store.HelixPropertyStore;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+
+
+/**
+ * A mock implementation of {@link AccountServiceFactory}. This is only for testing purpose and is not thread safe.
+ */
+public class MockHelixAccountServiceFactory extends HelixAccountServiceFactory {
+  private final HelixPropertyStoreConfig storeConfig;
+  private final AccountServiceMetrics accountServiceMetrics;
+  private final Notifier notifier;
+  private final boolean shouldUseMockHelixStore;
+  private final Map<String, MockHelixPropertyStore<ZNRecord>> storeKeyToMockStoreMap = new HashMap<>();
+
+  /**
+   * Constructor.
+   * @param verifiableProperties The properties to start a {@link HelixAccountService}.
+   * @param metricRegistry The {@link MetricRegistry} to start a {@link HelixAccountService}.
+   * @param notifier The {@link Notifier} to start a {@link HelixAccountService}.
+   * @param shouldUseMockHelixStore {@code true} if {@link HelixAccountService} is generated using a
+   *                                {@link MockHelixPropertyStore}; {@code false} using a real {@link HelixPropertyStore}.
+   */
+  public MockHelixAccountServiceFactory(VerifiableProperties verifiableProperties, MetricRegistry metricRegistry,
+      Notifier<String> notifier, boolean shouldUseMockHelixStore) {
+    super(verifiableProperties, metricRegistry, notifier);
+    storeConfig = new HelixPropertyStoreConfig(verifiableProperties);
+    accountServiceMetrics = new AccountServiceMetrics(metricRegistry);
+    this.notifier = notifier;
+    this.shouldUseMockHelixStore = shouldUseMockHelixStore;
+  }
+
+  @Override
+  public AccountService getAccountService() {
+    return new HelixAccountService(getHelixStore(storeConfig), accountServiceMetrics, notifier);
+  }
+
+  /**
+   * Gets a {@link HelixPropertyStore} configured by {@link #shouldUseMockHelixStore}, it will either return a
+   * {@link MockHelixPropertyStore} if set to {@code true}, or {@link HelixPropertyStore} if set to {@code false}.
+   * @param storeConfig The config for constructing a {@link HelixPropertyStore}.
+   * @return A {@link HelixPropertyStore}.
+   */
+  public HelixPropertyStore getHelixStore(HelixPropertyStoreConfig storeConfig) {
```

Comment:
It will return an existing store when the `shouldUseMockHelixStore` is set to `true`, but not for the `false` case, which will return a real `HelixPropertyStore` for every call. Have added into the javadoc as you suggest.